src/types.ts

```diff
@@ -282,7 +291,10 @@ export type BlockValueType =
   | CalloutValueType
   | BookmarkValueType
   | ToggleValueType
-  | CollectionValueType;
+  | CollectionValueType
+  | TweetType;
+
+export type BlockValueTypeKeys = Pick<BlockValueType, "type">["type"];
```

Comment:
This creates a type of all the "type" keys in `BlockValueType` so that we can properly type the customComponent prop.